File: src/main/scala/robosky/ether/HoeHacks.scala
package robosky.ether

import net.minecraft.block.{Block, BlockState}
import net.minecraft.item.{HoeItem, Item, ToolMaterials}

object HoeHacks extends HoeItem(ToolMaterials.WOOD, 0f, new Item.Settings()) {
  def addHoeable(source: Block, tgt: BlockState): BlockState = HoeItem.TILLED_BLOCKS.put(source, tgt)
}

File: src/main/java/robosky/uplands/HoeHacks.java
package robosky.uplands;

import net.minecraft.block.Block;
import net.minecraft.block.BlockState;
import net.minecraft.item.HoeItem;
import net.minecraft.item.Item;
import net.minecraft.item.ToolMaterials;

public final class HoeHacks extends HoeItem {

    private HoeHacks() {
        super(ToolMaterials.WOOD, 0f, new Item.Settings());
    }

    public static void addHoeable(Block source, BlockState tgt) {
        HoeItem.TILLED_BLOCKS.put(source, tgt);
    }
}
